ARK Server Manager/Lib/ServerRCON.cs

```diff
@@ -370,13 +370,45 @@ private void HandleCommand(ConsoleCommand command)
                     }
                 }).DoNotWait();
             }
+            else
+            {
+                var arkData = await ArkData.ArkDataContainer.CreateRemoteAsync();
+                arkData.SteamLoaded = true;
+                //await arkData.LoadSteamAsync(Config.Default.SteamAPIKey);
+                TaskUtils.RunOnUIThreadAsync(() =>
+                {
+                    foreach (var playerData in arkData.Players)
+                    {
+                        var playerToUpdate = this.Players.FirstOrDefault(p => p.SteamId == Int64.Parse(playerData.SteamId));
+                        if (playerToUpdate != null)
+                        {
+
+                            playerToUpdate.AvatarImage = new BitmapImage(new Uri(playerData.AvatarUrl));
```

Comment:
Unfortunately this pattern is very much against normal UI patterns and leads to frozen/choppy UI if the network calls take more than a few milliseconds.

The way to do this is the perform all of the network data acquisition on a background, non-UI thread.  When that thread completes, then you run a TaskUtils.RunOnUIThreadAsync() with the calls to update the UI elements.

So for example, you could do the following (note we do not await this call):

ArkData.ArkDataContainer.CreateRemoteAsync().ContinueWith(t => { var arkData = t.Result; ... remainder of function });

That means this method does not block waiting for the network operations to complete on the UI thread.  It also means the command will be issued and return before the data is likely collected.

Also, a bit odd that you are setting arkData.StreamLoaded from outside of the ArkData object - that does not seem like it is the responsibility of this code, but rather the ArkDataContainer's.

Does that make sense?  Getting the async stuff right along with the UI restrictions is a bit painful, but the payoff is that the UI doesn't freeze, especially if the user is experiencing network issues.